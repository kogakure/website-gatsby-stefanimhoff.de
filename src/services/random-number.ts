export const getRandomNumber = (max: number) =>
  String(Math.floor(Math.random() * Math.floor(max)));
