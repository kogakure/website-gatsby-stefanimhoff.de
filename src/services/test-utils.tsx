import React from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';

import { EmojifyProvider, LocalizedDateProvider } from '../contexts';

const customRender = (
  node: React.ReactElement,
  options?: RenderOptions
): RenderResult => {
  const rendered = render(
    <EmojifyProvider>
      <LocalizedDateProvider>{node}</LocalizedDateProvider>
    </EmojifyProvider>,
    options
  );

  return {
    ...rendered,
    rerender: (ui: React.ReactElement, rerenderOptions?: RenderOptions) =>
      customRender(ui, { container: rendered.container, ...options }),
  };
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
