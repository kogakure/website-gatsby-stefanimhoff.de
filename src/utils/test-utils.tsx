import React from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { EmojifyProvider, LocalizedDateProvider } from '../contexts';
import { ThemeType } from '../typings/Theme';
import { lightTheme } from '../themes';

interface CustomTheme {
  theme: ThemeType;
}

type RenderOptionsWithTheme = RenderOptions & CustomTheme;

const customRender = (
  node: React.ReactElement,
  options: RenderOptionsWithTheme = { theme: lightTheme }
): RenderResult => {
  const { theme } = options;
  const rendered = render(
    <ThemeProvider theme={theme}>
      <EmojifyProvider>
        <LocalizedDateProvider>{node}</LocalizedDateProvider>
      </EmojifyProvider>
    </ThemeProvider>,
    options
  );

  return {
    ...rendered,
    rerender: (
      ui: React.ReactElement,
      rerenderOptions?: RenderOptionsWithTheme
    ) => customRender(ui, { container: rendered.container, ...options }),
  };
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
