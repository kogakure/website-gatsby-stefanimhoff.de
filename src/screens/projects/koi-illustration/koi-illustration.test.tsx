import React from 'react';
import * as Gatsby from 'gatsby';
import { axe } from 'jest-axe';

import { render } from '../../../services/test-utils';

import { KoiIllustration } from '.';

const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery');

useStaticQuery.mockImplementation(() => ({
  koiIllustration: {
    childImageSharp: {
      fluid: {
        aspectRatio: 0.66,
        base64: 'data:image/jpeg;base64,XYZ',
        sizes: '(max-width: 1200px) 100vw, 1200px',
        src: '/static/cover.jpg',
        srcSet: '/static/cover.jpg 200w, /static/cover.jpg 400w',
      },
    },
  },
}));

describe('Projects', () => {
  describe('KoiIllustration', () => {
    test('renders correctly', () => {
      const { container } = render(<KoiIllustration />);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('is accessible', async () => {
      const { container } = render(<KoiIllustration />);
      const results = await axe(container);

      expect(results).toHaveNoViolations();
    });
  });
});
