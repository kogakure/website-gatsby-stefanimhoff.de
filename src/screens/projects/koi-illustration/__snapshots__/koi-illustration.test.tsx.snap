// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects KoiIllustration renders correctly 1`] = `
.c0 {
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  display: grid;
  grid-column: 1 / -1;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.c2 {
  border-color: var(--colorForegroundSoft);
  border-style: solid;
  border-width: 1px;
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-property: box-shadow,-webkit-transform;
  -webkit-transition-property: box-shadow,transform;
  transition-property: box-shadow,transform;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
}

.c2:hover,
.c2:focus {
  box-shadow: 0 0 50px rgba(0,0,0,0.2);
  -webkit-transform: scale(1.03);
  -ms-transform: scale(1.03);
  transform: scale(1.03);
}

.c3 {
  border-left-width: 0;
  border-right-width: 0;
}

@media screen and (prefers-reduced-motion:no-preference) {

}

@media (min-width:768px) {
  .c1 {
    grid-column: 1 / -6;
  }
}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

@media (min-width:768px) {
  .c3 {
    border-right-width: 1px;
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2 c3 gatsby-image-wrapper"
      style="position: relative; overflow: hidden;"
    >
      <div
        aria-hidden="true"
        style="width: 100%; padding-bottom: 151.5151515151515%;"
      />
      <img
        alt=""
        aria-hidden="true"
        src="data:image/jpeg;base64,XYZ"
        style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition-delay: 500ms;"
      />
      <picture>
        <source
          sizes="(max-width: 1200px) 100vw, 1200px"
          srcset="/static/cover.jpg 200w, /static/cover.jpg 400w"
        />
        <img
          alt=""
          loading="lazy"
          sizes="(max-width: 1200px) 100vw, 1200px"
          src="/static/cover.jpg"
          srcset="/static/cover.jpg 200w, /static/cover.jpg 400w"
          style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
        />
      </picture>
      <noscript>
        &lt;picture&gt;&lt;source srcset="/static/cover.jpg 200w, /static/cover.jpg 400w" sizes="(max-width: 1200px) 100vw, 1200px" /&gt;&lt;img loading="lazy" sizes="(max-width: 1200px) 100vw, 1200px" srcset="/static/cover.jpg 200w, /static/cover.jpg 400w" src="/static/cover.jpg" alt="" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
      </noscript>
    </div>
  </div>
</section>
`;
